src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -134,33 +214,42 @@ public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResu
 					continue;
 				}
 			}
-
 			variables.add(new NonNullPair<>(name, o));
 		}
+		getParser().getCurrentScript().addData(new DefaultVariables(variables));
 		return true;
 	}
 
 	@Override
 	public boolean load() {
-		for (NonNullPair<String, Object> pair : variables) {
+		DefaultVariables data = getParser().getCurrentScript().getData(DefaultVariables.class);
+		for (NonNullPair<String, Object> pair : data.getVariables()) {
 			String name = pair.getKey();
-			Object o = pair.getValue();
-
 			if (Variables.getVariable(name, null, false) != null)
 				continue;
 
-			Variables.setVariable(name, o, null, false);
+			Object value = pair.getValue();
+			Variables.setVariable(name, value, null, false);
 		}
 		return true;
 	}
 
+	@Override
+	public void postUnload() {
+		Script script = getParser().getCurrentScript();
+		DefaultVariables data = script.getData(DefaultVariables.class);
+		for (NonNullPair<String, Object> pair : data.getVariables())
+			Variables.setVariable(pair.getKey(), null, null, false);
+		script.removeData(DefaultVariables.class);
```

Comment:
Instead of doing this here I could maybe update https://github.com/SkriptLang/Skript/pull/5321 to clear the ScriptData after the postUnload phase. Minor functional difference than what is occurring here, but I don't think it would matter.

Reply:
I already suggested that to you. Unrelated to this pull currently.

Reply:
Sure. Once this is merged I'll make the changes and adapt here as needed.

Reply:
Sure. Once this is merged I'll make the changes and adapt here as needed.